### PR TITLE
Testing!

### LINKED ISSUE #77: Add CI test for installing packages / programs
From:

- [ ] GitHub Releases
- [ ]  zinit-packages monorepo
- [ ] Oh-My-zsh

### PR DESCRIPTION
**Pull request by pschmitt**
zunit testing, ahoy!

- [x] run tests locally via `scripts/zunit-tests.sh` (lefthook, here we come!)
- [x] annexes (2 at least)
- [x] binaries? (`sbin`). May be OOS.
- [x] plugins 
- [x] snippets (OMZP, OMZL and PZTM)
- [x] documentation (`doc/HACKING.md`)
- [x] update `doc/CHANGELOG.md`)
- [x] DRY setup/teardown
- [x] Add badge to `README.md`

This improves the docker image quite a bit:

- `ZINIT[HOME_DIR]` now points to `/data` which can be a volume (useful for CI testing w/ zunit)
- OMZL::key-bindings.zsh doesn't get downloaded at every startup

Fixes #77

Refs:
- #77 
- #42 
- #37 

~~Please squash when merging ☺️~~

### CODE REVIEW

**Comment by pschmitt** [comment]
This needs a follow-up to reach a higher test coverage (#testAllTheIces) but at least the framework is now there 🥳

**Comment by pschmitt** [comment]
The failing test (plugins) already highlights a currently unsolved issue that materializes when running `@zinit-scheduler-burst` with Ansible or other configuration management tools:

some ices get ignored when run non-interactively 🤷
Will open a separate issue for that.

File: doc/HACKING.md [hunk 1]
```diff
@@ -41,3 +41,73 @@ git checkout documentation
...
 [the README in the documentation]https://github.com/zdharma-continuum/zinit/blob/documentation/README.md
+
+
+# Testing
+
+We run out tests with [zunit](https://zunit.xyz).
+
+To add a new test case:
+
+1. Install [zunit]()https://zunit.xyz and [revolver](https://github.com/molovo/revolver):
```
Comment (alichtman) [inline doc/HACKING.md:52]:
```suggestion
1. Install [zunit](https://zunit.xyz) and [revolver](https://github.com/molovo/revolver):
```

**Review by alichtman — CHANGES_REQUESTED** [review]

**Comment by pschmitt** [comment]
The zunit plugin issue is fixed by #99

**Pull request merged**